### ISSUE TITLE
Make a release

### ISSUE THREAD

**Issue by fingolfin** [lbfm-rwth/carat#32]
This repository already contains a bunch of fixes compared to the last release. It would be nice if there was a release at some point, so that people can benefit from these fixes, even if there is more work to be done.

**Comment by DominikBernhardt** [comment]
I agree @fingolfin , I just don't have enough time at the moment. It's on my TODO list..

**Comment by fingolfin** [comment]
That's fine, but this way it's logged and I have a place to remind myself that I already bugged you about it :-).

BTW, this was motivated by https://github.com/gap-system/gap-distribution/issues/55 -- and addressing that will not just require a release from this repository, but presumably also one of https://github.com/gap-packages/carat right afterwards, with the new carat bundled in. (And perhaps the GAP package should be renamed??? I'll submit an issue there).

**Comment by DominikBernhardt** [comment]
Why do you think it should be renamed?

**Comment by gaehler** [comment]
There are indeed quite a bit of useful changes. In particular, the amount of compile time warnings has been drastically reduced. Also, the Carat binaries are now relocatable, and can find the location of the data base from an environment variable.

I would be very interested in making a new release of my GAP interface package and include an improved CARAT. I was however waiting for a new release of CARAT - so far in vain. One thing which is really missing is an updated documentation. The bundled documentation on GitHub is partly outdated (see issue #11), some parts don't work, and the official homepage in Aachen seems gone. There was a replacement at http://134.130.169.213/carat/index.html, which currently doesn't answer, however.

Of course, I could release my package bundled with a snapshot from GitHub, but would have preferred a real release.

**Comment by DominikBernhardt** [comment]
I agree with all of the above points, but I am currently busy with more important things, namely writing my PhD thesis and I don't have much time to devote to Carat unfortunately.

The website is not accessible anymore because there were some infrastructural changes in Aachen I had no influence on. I will move the webpage to GitHub pages eventually, but I have to extract to old html documentation from the old website (I can still access the old Webserver).

**Comment by DominikBernhardt** [comment]
@fingolfin : When should we make a release? Do you want other issues to be dealt with before we make one? Otherwise I guess we could release version `3.0` now?

**Comment by gaehler** [comment]
A few things that come to mind:
- we should add a configure script to the release branch and the tar ball, so that users do not need autotools. In parallel, we should also indicate in the README how the configure script is generated, in case it is missing.
- the new webpage should be mentioned in the REAMDE, documentation, etc.
- the corrections in doc/index.html on the webpage should also be ported to tex/index.html 

I guess the web page will not be part of the release, although it duplicates (and could replace) stuff in directory tex. This means that amendments to the web page can be made later, independently of the release. One thing that really should be changed there is the title. "GAP Package Carat" is really misleading. I'm not sure how hard-wired this is buried in the template.

**Comment by fingolfin** [comment]
I strongly disagree with putting a generated file, in particular a configure script, into the repository. That it belongs in release tarballs is standard practice, though.

The website template is fully customizable.
